src/pipeline.ts

```diff
@@ -482,14 +523,400 @@ export async function scanRssFeeds() {
 		});
 		await indexNewTorrents();
 		for (const [i, candidate] of candidatesSinceLastTime.entries()) {
+			const candidateLog = `${chalk.bold.white(candidate.name)} from ${candidate.tracker}`;
 			logger.verbose({
 				label: Label.RSS,
-				message: `Processing release ${i + 1}/${
-					candidatesSinceLastTime.length
-				}`,
+				message: `(${i + 1}/${candidatesSinceLastTime.length}) ${candidateLog}`,
 			});
-			await checkNewCandidateMatch(candidate, Label.RSS);
+			await checkNewCandidateMatch(candidate, candidateLog, Label.RSS);
 		}
 		logger.info({ label: Label.RSS, message: "Scan complete" });
 	}
 }
+
+export async function injectSavedTorrents() {
+	const { flatLinking, injectDir, linkDir, outputDir } = getRuntimeConfig();
+	const targetDir = injectDir ?? outputDir;
+	const targetDirLog = chalk.bold.magenta(targetDir);
+
+	if (injectDir) {
+		// injectDir defined only by `cross-seed inject`
+		logger.warn({
+			label: Label.INJECT,
+			message: `Manually injecting torrents performs minimal filtering which slightly increases chances of false positives, see the docs for more info`,
+		});
+	}
+	if (!fs.existsSync(targetDir) || !fs.statSync(targetDir).isDirectory()) {
```

Comment:
it'd be nice to consolidate this into one fs call where we catch ENOENTs rather than 2 separate calls. 

It may seem like this code protects against `ENOENT`, but if the file gets deleted in between the `existsSync` call and the `statSync` call, the program throws an unexpected error. By "begging for forgiveness" (catching ENOENT errors from `statSync`) rather than "asking for permission" (preflight `existsSync` calls), we harden the program against race conditions.